Enabling autogating in PyGRB

a-r-williamson opened the pull request:
These changes enable autogating in the PyGRB workflow.

titodalcanton commented:
Hi Andrew, I have a couple of comments on your change to `pycbc_condition_strain`:
* It looks like the `--gating-only` option will only work when `--frame-files` is used. If you have a chance, it would be useful to also support the `--frame-type` mode, as the `strain.from_cli()` function does (see https://github.com/ligo-cbc/pycbc/blob/master/pycbc/strain.py#L218).
* Please remove commented-out code and the added `logging.info(strain)` calls.
Thanks!

a-r-williamson commented:
@titodalcanton , thanks for your comments.

I added the gating-only option purely for purposes of running some tests to show during code review. I've removed that change for the final pull request.

Let me know if you have any concerns with the code as it is now.

titodalcanton commented:
Thanks @a-r-williamson, looks good as far as I'm concerned.

a-r-williamson commented:
@pannarale I have a comparison of the background with and without this autogating: [GRB 160111A backgrounds](https://geo2.arcca.cf.ac.uk/~c1239174/LVC/pygrb/ETMY_triggers_zoom.png)

For this extended period of bad glitchiness the autogating didn't gate the entire veto window, hence the presence of glitch-related triggers even with autogating. But it's what I would expect.

pannarale commented on pycbc/workflow/jobsetup.py:
```diff
@@ -1723,3 +1724,31 @@ class PycbcSplitBankXmlExecutable(PycbcSplitBankExecutable):
+                        extension=input_files[0].name.split('.', 1)[-1],
+                        tags=tags)
+        node.add_output_opt('--output-strain-file', out_file)
+
```
Hi @a-r-williamson, your comment made me think that it is wise to add
```
out_gates_file = File([…])
node.add_output_opt('--output-gates-file', out_gates_file)
```
here so that we can save the gating information to a textfile (out_gates_file).  This should then be stuck in the pyGRB output page to make it easier to spot glitch-related triggers we may gate even with autogating.

a-r-williamson replied:
Ok. I can add that quick fix. If you want it linked in the results page, that's a bigger change obviously.

pannarale replied:
Thanks, @a-r-williamson. We can worry about updating the results page in a separate pull request (it is a somewhat separate issue after all).

pannarale commented:
Can someone with the right permissions approve this pull request and merge?

a-r-williamson commented:
@titodalcanton Do you have permission to review and approve this?

pannarale commented:
@a-r-williamson, @titodalcanton, do you get a Server Error (500) from the code-quality/landscape link?

titodalcanton commented:
And yes, I also get a 500 error. Let's wait a couple hours I guess?
